src/box/lua/config/applier/credentials.lua

```diff
@@ -296,6 +296,62 @@ local function privileges_add_defaults(name, role_or_user, intermediate)
     return res
 end
 
+--
+-- Create a credential sharding role if it is assigned to any user or role and
+-- has not been declared in 'credentials.roles'.
+--
+-- Replicasets with the sharding storage role must be assigned the credentials
+-- sharding role. In these replicasets, the role will be assigned all necessary
+-- privileges.
+--
+-- For replicasets that do not have the sharding storage role assigned, the
+-- credentials sharding role will not have any privileges.
```

Comment:
Nit: We discussed it as a workaround solution for problems around ER_SCHEMA_NEEDS_UPGRADE. I think it should be clarified here. So, we'll have this information if we'll go to refactor config-vshard interface.

Reply:
Expanded the comment.

Reply:
Nice, thanks!